thewalrus/_permanent.py

```diff
@@ -54,6 +54,12 @@ def perm(A, method="bbfg"):
     if np.isnan(A).any():
         raise ValueError("Input matrix must not contain NaNs.")
 
+    if matshape[0] == 0:
+        return 1
```

Comment:
It could be nice to return the same type as the input NumPy array (e.g., `1+0j` if it's complex). Same below.

Reply:
I do this now, though just a note that none of the other functions like the Hafnian are implementing this logic, though it does in general seem a better more consistent approach.